src/opnsense/mvc/app/library/OPNsense/Mvc/Request.php

```diff
@@ -0,0 +1,136 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Mvc;
+
+use OPNsense\Core\SanitizeFilter;
+use stdClass;
+
+class Request
+{
+    private string $rawBody = '';
+
+    /**
+     * @param string $header retrieve request header by name
+     * @return string content
+     */
+    public function getHeader(string $header): string
+    {
+        $name = strtoupper(strtr($header, "-", "_"));
+        if (isset($_SERVER[$name])) {
+            return $_SERVER[$name];
+        } elseif (isset($_SERVER["HTTP_$name"])) {
+            return $_SERVER["HTTP_$name"];
+        }
+        return '';
+    }
+
+    /**
+     * @return string method name (GET, POST, PUT, ...)
+     */
+    public function getMethod(): string
+    {
+        return $_SERVER["REQUEST_METHOD"];
+    }
+
+    public function isPost(): bool
+    {
+        return $this->getMethod() == 'POST';
+    }
+
+    public function isGet(): bool
+    {
+        return $this->getMethod() == 'GET';
+    }
+
+    public function isPut(): bool
+    {
+        return $this->getMethod() == 'PUT';
+    }
+
+    public function isDelete(): bool
+    {
+        return $this->getMethod() == 'DELETE';
+    }
+
+    public function isHead(): bool
+    {
+        return $this->getMethod() == 'HEAD';
+    }
+
+    public function isPatch(): bool
+    {
+        return $this->getMethod() == 'PATCH';
+    }
+
+    public function isOptions(): bool
+    {
+        return $this->getMethod() == 'OPTIONS';
+    }
+
+    public function getRawBody(): string
+    {
+        if (empty($this->rawBody)) {
+            $this->rawBody = file_get_contents("php://input");
+        }
+        return $this->rawBody;
+    }
+
+    public function has(string $name): bool
+    {
+        return isset($_REQUEST[$name]);
+    }
+
+    public function hasPost(string $name): bool
+    {
+        return isset($_POST[$name]);
+    }
+
+    public function getPost(string $name, ?string $filter = null, ?string $defaultValue = null)
+    {
+        $value = isset($_POST[$name]) ? $_POST[$name] : $defaultValue;
+        if ($filter !== null && $value !== null) {
+            $value = (new SanitizeFilter())->sanitize($value, $filter);
+        }
+        return $value;
+    }
+
+    public function get(string $name, ?string $filter = null, ?string $defaultValue = null)
+    {
+        $value = isset($_REQUEST[$name]) ? $_REQUEST[$name] : $defaultValue;
+        if ($filter !== null) {
+            $value = (new SanitizeFilter())->sanitize($value, $filter);
+        }
+        return $value;
+    }
+
+    public function getJsonRawBody(bool $associative = false): stdClass| array| bool
+    {
+        return json_decode($this->getRawBody(), $associative) ?? false;
```

Comment:
Feels like this can be simplified in the future. I've never seen us use JSON as an object and likely will not happen anytime soon.

Reply:
only caller seems to be ApiControllerBase, I'll remove the parameter as it's highly unlikely anyone else uses this (a grep on plugins shows nothing)